[IMP] sales: updated Amazon Connector setup doc for 16 (#6288)
odoo/documentation: content/applications/sales/sales/amazon_connector/setup.rst

.. note::
   After validation of credentials, the marketplaces to which this account has access is
   synchronized, and automatically made available.

After saving, the field in the :guilabel:`Credentials` tab is replaced by a :guilabel:`Link with
Amazon` button.

.. image:: setup/amazon-accounts-form-link-button.png
   :align: center
   :alt: A typical Amazon Account form page and Link with Amazon button in Odoo Sales.

Clicking that button redirects to either the Amazon login page, or directly to the required consent
page, if the user is already logged in to Amazon.

On this page, confirm that Amazon is allowed to give Odoo access to the account and related data.

Upon confirmation, Amazon returns the user to Odoo, and the account has been registered.

With the account successfully registered, the marketplaces available to this specific account are
synchronized and listed under the :guilabel:`Marketplaces` tab.

If desired, remove items from the list of synchronized marketplaces to disable synchronization.

Match database products in Amazon
=================================

When an Amazon order is synchronized, up to three sales order items are created in Odoo for each

> product sold on Amazon: one for the marketplace product, one for the shipping charges (if any), and

Reviewer: @Demesmaeker can I have you opinion on this ? Not sure it is clear and not sure it reflects how it works exaclty

Reviewer 2: So yes, it's correct. I can't think of a clearer way to say it though. 1 to 3 sale order line per product, 1 line for the product, 1 for the delivery, 1 for the gift wrapping charge and/or the message. =]

one for the gift wrapping charges (if any).

.. _amazon/matching:

The selection of a database product for a sales order item is done by matching its
:guilabel:`Internal Reference` with the *SKU* for marketplace items, the *Shipping Code* for
delivery charges, and the *Gift Wrapping* code for gift wrapping charges.

For marketplace products, pairings are saved as *Amazon Offers*, which are listed under the

> :guilabel:`Offers` button on the account form.

Reviewer: @ksc-odoo maybe we can add an image to point it out

Reviewer: @ksc-odoo Amazon does not provide the possibility of creating fake account unfortunately.
I will provide you some pictures (don't hesitate to ask for more if you want)

Author: @dalaOdoo @Demesmaeker I added a screenshot of this (from the shared google drive of Amazon Tour images). Please give this a quick review and let me know if it's good enough to pass on to the next review stage. Thanks! 👍

==============================

> Amazon Connector configuration

Author: Okay, I added a brief intro paragraph detailing those mentioned elements. I think it provides a lot more (necessary) context to lead into the rest of the doc. Thanks for the awesome suggestion!

   :align: center
   :alt: A typical Amazon Account form page in the Odoo Sales application.

.. note::
   After validation of credentials, the marketplaces to which this account has access is
   synchronized, and automatically made available.

After saving, the field in the :guilabel:`Credentials` tab is replaced by a :guilabel:`Link with
Amazon` button.

.. image:: setup/amazon-accounts-form-link-button.png
   :align: center
   :alt: A typical Amazon Account form page and Link with Amazon button in Odoo Sales.

Clicking that button redirects to either the Amazon login page, or directly to the required consent
page, if the user is already logged in to Amazon.

On this page, confirm that Amazon is allowed to give Odoo access to the account and related data.

Upon confirmation, Amazon returns the user to Odoo, and the account has been registered.

With the account successfully registered, the marketplaces available to this specific account are
synchronized and listed under the :guilabel:`Marketplaces` tab.

If desired, remove items from the list of synchronized marketplaces to disable synchronization.

Match database products in Amazon
=================================

> When an Amazon order is synchronized, up to three sales order items are created in Odoo for each

Author: I totally understand where you're coming from. I always read it as up to 3 line items being added to the sales order in Odoo, to represent the 3 potential charges coming from the Amazon purchase. As you know, I was only able to work off context clues provided in the previous version of this doc, because I don't have access to a paid-for Amazon Seller account. However, since it was previously "OK'd" by the PO/PE, I think I just need to clarify this sentence a bit more (which I just took care of :) ) -- I believe the newly restructured sentence spells things out a bit clearer. That said, if you think it should be changed even further, I'm always open to suggestions! Thanks for pointing this out!

product sold on Amazon: one for the marketplace product, one for the shipping charges (if any), and
one for the gift wrapping charges (if any).

.. _amazon/matching:

The selection of a database product for a sales order item is done by matching its
:guilabel:`Internal Reference` with the *SKU* for marketplace items, the *Shipping Code* for

> delivery charges, and the *Gift Wrapping* code for gift wrapping charges.

Author: Great question. So, from what I gathered from the last doc on this, I do, indeed, believe the SKU, Shipping Code, and Gift Wrapping code are Amazon-related elements. I went ahead and clarified this a bit to avoid any confusion. I also added a quick explanation/reminder as to what an Internal Reference is in Odoo. I believe this should remedy the situation. However, if you think additional adjustments need to be made, just let me know. Thanks again for bringing this to my attention!

For marketplace products, pairings are saved as *Amazon Offers*, which are listed under the
:guilabel:`Offers` button on the account form.

.. image:: setup/amazon-offers-button.png
   :align: center
   :alt: The Amazon Offers smart button on the account form in Odoo Sales.

Offers are automatically created when the pairing is established, and they're used for subsequent
orders to lookup SKUs. If no offer with a matching SKU is found, :ref:`the internal reference is
used instead <amazon/matching>`.

.. tip::
   It's possible to force the pairing of a marketplace item with a specific product, by changing

>    either the product or the SKU of an offer. The offer can be manually created if it was not

Reviewer 3: What are the steps to do this?

Author: Again, if I had access to an actual Amazon Seller account on Marketplace, I'd likely be able to break down the steps required to do this. However, given that I am only able to gather what I can from previously-written documentation context, I *believe* this is, essentially, suggesting that the Product Name or the SKU can be directly edited to ensure they match in this instance. I added a brief mention of that to the end of the sentence, which (I hope) clears things up a bit. But, if you think this still needs a bit of work, just let me know, and I'll happily change it to whatever you deem best. Thanks!

Reviewer: From amazon account you can find the list of offers with this smart button
![image](https://github.com/odoo/documentation/assets/100136841/339ed233-c61c-4ad3-805b-60df38fbd88e)

Then you can create, or edit offer to map odoo product with amazon SKU
![image](https://github.com/odoo/documentation/assets/100136841/a31aa211-6f19-46e6-859d-9adc1b66f301)

   automatically done yet.

   This is useful if the internal reference is not used as the SKU, or if the product sells under
   different conditions.

> If no database product with a matching internal reference is found for a given SKU, or gift wrapping
> code, a default database product, *Amazon Sale*, is used. The same is done with the default product,
> *Amazon Shipping* and the shipping code.

Reviewer 3: Clarifying the language here
```suggestion
If no database product with a matching internal reference is found for a given Amazon SKU or gift wrapping
code, then a default database product, *Amazon Sale*, is used. The same is done with the default product
*Amazon Shipping* if no database product is found for a given Amazon shipping code.
```

.. note::
   To modify default products, activate the :ref:`developer mode <developer-mode>`, and navigate
   to :menuselection:`Sales app --> Configuration --> Settings --> Connectors --> Amazon Sync -->
   Default Products`.

Product tax configuration
=========================

To allow for tax reporting of Amazon sales with Odoo, the taxes applied to the sales order items are
those set on the product, or determined by the fiscal position.

Make sure to have the correct taxes set on your products in Odoo, or have it done by a fiscal

> position, to avoid discrepancies in the subtotals between *Seller Central* and Odoo.

Reviewer 3: ```suggestion
position, to avoid discrepancies in the subtotals between *Amazon Seller Central* and Odoo.
```

.. note::
   As Amazon does not necessarily apply the same taxes as those configured in Odoo, it may happen

>    that order totals differ by a few cents on *Seller Central*. Those differences can be resolved

Reviewer 3: ```suggestion
   that order totals differ by a few cents between Odoo and *Amazon Seller Central*. Those differences can be resolved
```

   with a write-off when reconciling the payments in Odoo.

.. _amazon/add-new-marketplace:

Add a new marketplace
=====================

:ref:`All marketplaces are supported by the Amazon Connector <amazon/supported-marketplaces>`, but
recently created ones might be missing from the database.

To add a new marketplace, proceed as follows:

> #. Activate the :ref:`developer mode <developer-mode>` (:menuselection:`Settings app --> Activate
>    the developer mode`.

Reviewer 3: ```suggestion
#. Activate the :ref:`developer mode <developer-mode>`.
```

#. Go to :menuselection:`Sales app --> Configuration --> Settings --> Connectors --> Amazon Sync -->
   Amazon Marketplaces`.
#. Click :guilabel:`New` to create a new marketplace record.
#. Enter the Marketplace ID in the :guilabel:`API Idenifier` field, and select the :guilabel:`Amazon
   Region` for your marketplace as described in the `Amazon Documentation for marketplace IDs and
   regions <https://developer-docs.amazon.com/amazon-shipping/docs/marketplace-ids>`_, and the
   :guilabel:`Seller Central URL` as described in the `Amazon Documentation for seller central URLs
   <https://developer-docs.amazon.com/amazon-shipping/docs/seller-central-urls>`_.

> #. Set the name of the record to `Amazon.<country code>` to easily retrieve it (e.g.:

Reviewer 3: ```suggestion
#. Set the name of the record to `Amazon.<country code>` to easily retrieve it (e.g.
```

   `Amazon.se`). The :guilabel:`API Identifier`, the :guilabel:`Region` and the :guilabel:`Seller

>    Central URL` fields should respectively hold the *Marketplace Id*, the selected Amazon region,

Reviewer 3: ```suggestion
   Central URL` fields should respectively hold the *Marketplace ID*, the selected Amazon region,
```

   and the *Seller Central URL* values from the Amazon Documentation.
#. Once the marketplace is saved, update the Amazon Account configuration by going to
   :menuselection:`Sales app --> Configuration --> Settings --> Connectors --> Amazon Sync -->
   Amazon Accounts`.

> #. Open the account on which to use the new marketplace, go to the :guilabel:`Marketplaces` tab, and

Reviewer 3: ```suggestion
#. Select the account on which to use the new marketplace, go to the :guilabel:`Marketplaces` tab, and
```

==============================
Amazon Connector configuration
==============================

Odoo allows users to register an Amazon seller account in the database, but the user **must** have
a paid Amazon Seller account prior to completing the configuration.

To learn how to set up a paid Seller account on Amazon, log in to Amazon, and select *Start a
Selling Account* from the *Account & Lists* drop-down menu, located in the header. Once that process
is complete, proceed to follow the instructions below to register and link that Amazon Seller

> account in Odoo.

Reviewer 4: Is there any Amazon documentation that we can link to here in this sentence? Or will they just have to log into Amazon and see for themselves?

Author: Yeah, unfortunately, they would have to log into Amazon and go from there. But I fully understand where you're coming from.

> To learn how to set up a paid Seller account on Amazon, log in to Amazon, and select *Start a
> Selling Account* from the *Account & Lists* drop-down menu, located in the header. Once that process
> is complete, proceed to follow the instructions below to register and link that Amazon Seller
> account in Odoo.

Reviewer 5: Specificity, wording, missing gui/menu labels.

Here's a rewrite with an added **seealso** hyperlink to make it easier for the user to sign up for a Seller account.

```suggestion
Set up a paid Seller account on Amazon by first logging into the Amazon platform, and navigating to
:menuselection:`Account \& Lists --> Start a Selling Account` from the drop-down menu located in
the header section.

Then on the :guilabel:`Sell with Amazon` page, follow the sign-up process and finally proceed to
follow the instructions below to register and link that Amazon Seller account in Odoo.

.. seealso::
   `Sell with Amazon <https://www.amazon.com/b/?node=12766669011>`
```

> Register Amazon account in Odoo
> ===============================

Reviewer 5: It's not a registration like signing up for a free trial, an Amazon Seller account, etc.

```suggestion
Connect Amazon Seller account to Odoo
=====================================
```

.. _amazon/setup:

> To register a seller account in Odoo, navigate to :menuselection:`Sales app --> Configuration
> --> Settings --> Connectors section`, activate the :guilabel:`Amazon Sync` feature, and click

Reviewer 5: ```suggestion
To register a seller account in Odoo, navigate to :menuselection:`Sales app --> Configuration -->
Settings --> Connectors section`, activate the :guilabel:`Amazon Sync` feature, and click
```

> To register a seller account in Odoo, navigate to :menuselection:`Sales app --> Configuration -->
> Settings --> Connectors section`, activate the :guilabel:`Amazon Sync` feature, and click
> :guilabel:`Save`.

Reviewer 5: ```suggestion
To connect an Amazon seller account in Odoo, navigate to :menuselection:`Sales app -->
Configuration --> Settings --> Connectors section`, activate the :guilabel:`Amazon Sync` feature,
and click :guilabel:`Save`.
```

Then, return to :menuselection:`Sales app --> Configuration --> Settings --> Connectors section`,
and click on the :guilabel:`Amazon Accounts` link under the :guilabel:`Amazon Sync` setting.

.. image:: setup/amazon-accounts-link-setting.png
   :align: center

>    :alt: The Amazon Accounts link beneath the Amazon Synch settings in Odoo Sales.

Reviewer 4: ```suggestion
   :alt: The Amazon Accounts link beneath the Amazon Sync settings in Odoo Sales.
```
Right?

Doing so reveals a separate :guilabel:`Amazon Accounts` page. From here, click :guilabel:`New` to

> create, and subsequently link, a new Amazon account.

Reviewer 4: ```suggestion
create and link a new Amazon account.
```

the seller account was initially created from the :guilabel:`Home Marketplace` drop-down menu.

.. image:: setup/amazon-accounts-form-page.png
   :align: center
   :alt: A typical Amazon Account form page in the Odoo Sales application.

After saving, the field in the :guilabel:`Credentials` tab is replaced by a :guilabel:`Link with
Amazon` button.

.. image:: setup/amazon-accounts-form-link-button.png
   :align: center
   :alt: A typical Amazon Account form page and Link with Amazon button in Odoo Sales.

Clicking that button redirects to either the Amazon login page, or directly to the required consent
page, if the user is already logged in to Amazon.

On the login page, log in to the desired Amazon seller account.

On the consent page, confirm that Amazon is allowed to give Odoo access to the account and related
data.

Upon confirmation, Amazon returns the user to Odoo, and the account has been registered.

With the Amazon account successfully registered, the marketplaces available to this specific account
are synchronized with Odoo and listed under the :guilabel:`Marketplaces` tab.

If desired, remove items from the list of synchronized marketplaces to disable synchronization.

> Match database products in Amazon
> =================================

Reviewer 5: Nowhere in this section do we "match database products in Amazon"

Here's a more appropriate heading for this section.

```suggestion
Amazon orders in Odoo
=====================
```

> When an Amazon order is synchronized, up to three line items are created on the sales order in Odoo.

Reviewer 5: Q: does this happen automatically via the **Connector** or is there a manual sync that can be performed? It's not clear *how* orders are synchronized...

For marketplace products, pairings are saved as *Amazon Offers*, which are listed under the
:guilabel:`Offers` smart button on the account form.

.. image:: setup/amazon-offers-button.png
   :align: center
   :alt: The Amazon Offers smart button on the account form in Odoo Sales.

Offers are automatically created when the pairing is established, and they're used for subsequent
orders to lookup SKUs. If no offer with a matching SKU is found, :ref:`the internal reference is
used instead <amazon/matching>`.

.. tip::
   It's possible to force the pairing of a marketplace item with a specific product, by changing
   either the product or the SKU of an offer to ensure they match. The offer can be manually created
   if it was not automatically done yet.

   This is useful if the internal reference is not used as the SKU, or if the product sells under
   different conditions.

If no database product with a matching internal reference is found for a given Amazon SKU or gift
wrapping code, then a default database product, *Amazon Sale*, is used. The same is done with the
default product *Amazon Shipping* if no database product is found for a given Amazon shipping code.

.. note::

>    To modify default products, activate the :ref:`developer mode <developer-mode>`, and navigate
>    to :menuselection:`Sales app --> Configuration --> Settings --> Connectors --> Amazon Sync -->

Reviewer 5: ```suggestion
   To modify default products, activate the :ref:`developer mode <developer-mode>`, and navigate to
   :menuselection:`Sales app --> Configuration --> Settings --> Connectors --> Amazon Sync -->
```

   Default Products`.

> Product tax configuration

Reviewer 5: There's no actual instruction here in this section.

Ideally there'd be links to Accounting articles, as well, to further help the reader in this use case, e.g. setting taxes on a product form or determining a fiscal position.

=========================

To allow for tax reporting of Amazon sales with Odoo, the taxes applied to the sales order items are
those set on the product, or determined by the fiscal position.

Make sure to have the correct taxes set on your products in Odoo, or have it done by a fiscal
position, to avoid discrepancies in the subtotals between *Amazon Seller Central* and Odoo.

.. note::
   As Amazon does not necessarily apply the same taxes as those configured in Odoo, it may happen
   that order totals differ by a few cents between Odoo and *Amazon Seller Central*. Those
   differences can be resolved with a write-off when reconciling the payments in Odoo.

.. _amazon/add-new-marketplace:

Add a new marketplace
=====================

> :ref:`All marketplaces are supported by the Amazon Connector <amazon/supported-marketplaces>`, but

Reviewer 5: Unnecessary ref.
It just points to the same sentence on another page so we had the reader click away from the page to read the same content twice 😁 

Better to just do this imo
```suggestion
All marketplaces are supported by the Amazon Connector. To add a new marketplace, proceed as
follows:
```

recently created ones might be missing from the database.

To add a new marketplace, proceed as follows:

#. Activate the :ref:`developer mode <developer-mode>`.
#. Go to :menuselection:`Sales app --> Configuration --> Settings --> Connectors --> Amazon Sync -->
   Amazon Marketplaces`.
#. Click :guilabel:`New` to create a new marketplace record.
#. Enter the Marketplace ID in the :guilabel:`API Idenifier` field, and select the :guilabel:`Amazon
   Region` for your marketplace as described in the `Amazon Documentation for marketplace IDs and
   regions <https://developer-docs.amazon.com/amazon-shipping/docs/marketplace-ids>`_, and the
   :guilabel:`Seller Central URL` as described in the `Amazon Documentation for seller central URLs
   <https://developer-docs.amazon.com/amazon-shipping/docs/seller-central-urls>`_.

> #. Set the name of the record to `Amazon.<country code>` to easily retrieve it (e.g. `Amazon.se`).

Reviewer 5: ```suggestion
#. Set the :guilabel:`Name` of the record to `Amazon.<country code>` to easily retrieve it (e.g. `Amazon.se`).
```

> #. Set the name of the record to `Amazon.<country code>` to easily retrieve it (e.g.
>    `Amazon.se`). The :guilabel:`API Identifier`, the :guilabel:`Region` and the :guilabel:`Seller
>    Central URL` fields should respectively hold the *Marketplace ID*, the selected Amazon region,
>    and the *Seller Central URL* values from the Amazon Documentation.

Reviewer 5: ```suggestion
#. Set the name of the record to `Amazon.<country code>` to easily retrieve it (e.g. `Amazon.se`).
   The :guilabel:`API Identifier`, the :guilabel:`Region` and the :guilabel:`Seller Central URL`
   fields should respectively hold the *Marketplace ID*, the selected Amazon region, and the *Seller
   Central URL* values from the Amazon Documentation.
```

#. Once the marketplace is saved, update the Amazon Account configuration by going to
   :menuselection:`Sales app --> Configuration --> Settings --> Connectors --> Amazon Sync -->
   Amazon Accounts`.
#. Select the account on which to use the new marketplace, go to the :guilabel:`Marketplaces` tab,

>    and click on :guilabel:`Update available marketplaces` (an animation should confirm the success
>    of the operation). Newly added marketplaces are automatically added to the list of synchronized

Reviewer 4: ```suggestion
   and click on :guilabel:`Update available marketplaces`. An animation should confirm the success
   of the operation. Newly added marketplaces are automatically added to the list of synchronized
```

   marketplaces. If the new marketplace is not added to the list, it means that it is either

>    incompatible or unavailable for the seller account.

Reviewer 4: Any alternative solutions in this case, if someone runs into this?

Author: Again, without proper access to a paid-for Amazon Seller account to create this documentation, the depth of topics is a bit limited. So, if they run into an error at this point, they'll have to reach out to support or figure out a solution on their own. That said, if we get comments requesting such a solution, we can reach out to POs for more info. But, I believe, for the time being, this can stand as-is. Thanks for pointing it out, though.

--> Settings --> Connectors section`, activate the :guilabel:`Amazon Sync` feature, and click
:guilabel:`Save`.

Then, return to :menuselection:`Sales app --> Configuration --> Settings --> Connectors section`,
and click on the :guilabel:`Amazon Accounts` link under the :guilabel:`Amazon Sync` setting.

.. image:: setup/amazon-accounts-link-setting.png
   :align: center
   :alt: The Amazon Accounts link beneath the Amazon Synch settings in Odoo Sales.

Doing so reveals a separate :guilabel:`Amazon Accounts` page. From here, click :guilabel:`New` to
create, and subsequently link, a new Amazon account.

On the blank :guilabel:`Amazon Account` form page, start by choosing a name for the account (e.g.
`American Marketplace`). Then, in the :guilabel:`Credentials` tab, select the marketplace on which
the seller account was initially created from the :guilabel:`Home Marketplace` drop-down menu.

.. image:: setup/amazon-accounts-form-page.png
   :align: center
   :alt: A typical Amazon Account form page in the Odoo Sales application.

.. note::
   After validation of credentials, the marketplaces to which this account has access are

>    synchronized, and automatically made available.

Reviewer 2: Isn't it strange to have this note before explaining how to validate the credentials? Especially since it's clearly written after, outside a note (L61-62) :thinking:

Author: Good call. I'll remove this note right away. Thanks for pointing this out. 👍
